src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateRepresentationService.java

```diff
@@ -0,0 +1,159 @@
+    private final ChangeOfRepresentationService changeOfRepresentationService;
+
+    private static final String CHANGE_REQUEST_FIELD = "changeOrganisationRequestField";
+    private static final String NOC_EVENT = "nocRequest";
+    private static final String CHANGE_OF_REPS = "ChangeOfRepresentatives";
+
+    private boolean isApplicant;
+
+    public Map<String, Object> updateRepresentationAsSolicitor(CaseDetails caseDetails,
+                                                               String authToken)  {
+
+        log.info("Updating representation for case ID {}", caseDetails.getId());
+
+        AuditEvent auditEvent = auditEventService.getLatestAuditEventByName(caseDetails.getId().toString(), NOC_EVENT)
+            .orElseThrow(() -> new IllegalStateException(String.format("Could not find %s event in audit", NOC_EVENT)));
+
+        UserDetails solicitorToAdd = idamClient.getUserByUserId(authToken, auditEvent.getUserId());
+        ChangeOrganisationRequest change = getChangeOrganisationRequest(caseDetails);
+        isApplicant = change.getCaseRoleId().getValueCode().equals(APP_SOLICITOR_POLICY);
+
+        ChangedRepresentative addedSolicitor = ChangedRepresentative.builder()
+            .name(solicitorToAdd.getFullName())
+            .email(solicitorToAdd.getEmail())
+            .organisation(change.getOrganisationToAdd())
+            .build();
+
+        ChangedRepresentative removedSolicitor = Optional.ofNullable(change.getOrganisationToRemove())
+            .map(org -> ChangedRepresentative.builder()
+            .name(isApplicant ? getApplicantSolicitorName(caseDetails)
+                : (String) caseDetails.getData().get(RESP_SOLICITOR_NAME))
+            .email(isApplicant ? getApplicantSolicitorEmail(caseDetails)
+                : (String) caseDetails.getData().get(RESP_SOLICITOR_EMAIL))
+            .organisation(org).build())
+            .orElse(null);
+
+        log.info("About to start updating solicitor details in the case data for caseId: {}", caseDetails.getId());
+        caseDetails.getData().putAll(updateCaseDataWithNewSolDetails(caseDetails, addedSolicitor));
+
+        return updateChangeOfRepresentatives(caseDetails, addedSolicitor, removedSolicitor);
+    }
+
+    private Map<String, Object> updateChangeOfRepresentatives(CaseDetails caseDetails,
+                                                              ChangedRepresentative addedSolicitor,
+                                                              ChangedRepresentative removedSolicitor) {
+
+        Map<String, Object> caseData = caseDetails.getData();
+        ChangeOfRepresentatives current = ChangeOfRepresentatives.builder()
+            .changeOfRepresentation(objectMapper.convertValue(caseData.get(CHANGE_OF_REPS),
+                new TypeReference<>() {}))
+            .build();
+
+        ChangeOfRepresentatives change = changeOfRepresentationService.generateChangeOfRepresentatives(
```

Comment:
same here, loads happening, a method should be focused on as few tasks as possible, ideally one task, if more behaviour is needed, delegate to other smaller methods